src/hotspot/share/classfile/classLoader.cpp

```diff
@@ -285,9 +286,9 @@ ClassPathZipEntry::~ClassPathZipEntry() {
   FREE_C_HEAP_ARRAY(char, _zip_name);
 }
 
-u1* ClassPathZipEntry::open_entry(const char* name, jint* filesize, bool nul_terminate, TRAPS) {
+u1* ClassPathZipEntry::open_entry(const char* name, jint* filesize, bool nul_terminate) {
     // enable call to C land
-  JavaThread* thread = THREAD->as_Java_thread();
+  JavaThread* thread = JavaThread::current();
```

Comment:
Is there a reason you didn't continue to pass in the thread rather than having to reify it again with JavaThread::current()?

Reply:
Because the caller of the above function doesn't need the thread. I've changed my patch so that the thread will be passed along as `Thread* current`.